python/lsst/ts/mtdome/mock_llc/amcs.py

```diff
@@ -267,13 +296,88 @@ async def fans(self, action: str) -> float:
+            The TAI time, unix seconds, when the command was issued. To model
+            the real dome, this should be the current time. However, for unit
+            tests it can be convenient to use other values.
+
+        Returns
+        -------
+        `float`
+            The expected duration of the command.
+        """
+        self.duration = self.azimuth_motion.exit_fault(start_tai)
+        return self.duration
+
+    async def reset_drives_az(self, start_tai: float, reset: typing.List[int]) -> float:
+        """Reset one or more AZ drives.
+
+        Parameters
+        ----------
+        start_tai: `float`
+            The TAI time, unix seconds, when the command was issued. To model
+            the real dome, this should be the current time. However, for unit
+            tests it can be convenient to use other values.
+        reset: array of int
+            Desired reset action to execute on each AZ drive: 0 means don't
+            reset, 1 means reset.
+
+        Returns
+        -------
+        `float`
+            The expected duration of the command.
+
+        Notes
+        -----
+        This is necessary when exiting from FAULT state without going to
+        Degraded Mode since the drives don't reset themselves.
+        The number of values in the reset parameter is not validated.
+        """
+        self.duration = self.azimuth_motion.reset_drives_az(start_tai, reset)
+        return self.duration
+
+    async def calibrate_az(self, start_tai: float) -> float:
+        """Take the current position of the dome as zero. This is necessary as
+        long as the racks and pinions on the drives have not been installed yet
+        to compensate for slippage of the drives.
+
+        Returns
+        -------
+        `float`
+            The expected duration of the command.
+
         Parameters
         ----------
         start_tai: `float`
             The TAI time, unix seconds, when the command was issued. To model
             the real dome, this should be the current time. However, for unit
             tests it can be convenient to use other values.
         """
-        self.azimuth_motion.exit_fault(start_tai)
+        self.azimuth_motion.calibrate_az(start_tai)
         self.duration = 0.0
         return self.duration
+
+    async def set_fault(
+        self, start_tai: float, drives_in_error: typing.List[int]
+    ) -> None:
+        """Set the LlcMotionState of AMCS to fault and set the drives in
+        drives_in_error to error.
+
+        Parameters
+        ----------
+        start_tai: `float`
+            The TAI time, unix seconds, when the command was issued. To model
+            the real dome, this should be the current time. However, for unit
+            tests it can be convenient to use other values.
+        drives_in_error : array of int
+            Desired error action to execute on each AZ drive: 0 means don't
+            set to error, 1 means set to error.
```

Comment:
Please consider booleans, unless you anticipate more than one error state, in which case use an enum.

Reply:
This also applies to the mock implementation of the real dome software and I unfortunately cannot make changes like that without consulting the vendor first.